src/Functions/FunctionSQLJSON.h

```diff
@@ -247,10 +275,7 @@ class JSONValueImpl
         {
             if (status == VisitorStatus::Ok)
             {
-                if (!(current_element.isArray() || current_element.isObject()))
-                {
-                    break;
-                }
```

Comment:
This changes previous behaviour of the function, it can be unexpected. Let's also do it under a setting.